src/Logary/Targets_Core.fs

```diff
@@ -108,6 +108,234 @@ module TextWriter =
+  type LiterateConsoleConf =
+    { formatProvider    : IFormatProvider
+      getLogLevelText   : LogLevel -> string
+      tokenise          : LiterateConsoleConf -> Message -> (string * LiterateToken) list 
+      theme             : LiterateToken -> ConsoleColor
+      colourWriter      : obj -> (string * ConsoleColor) list -> unit }
+
+  module internal LiterateFormatting =
+    open System.Text
+    open Logary.Utils.FsMessageTemplates
+
+    let literateExceptionColorizer (options : LiterateConsoleConf) (typeName) (message) (stackTrace) =
+      let stackFrameLinePrefix = "   "
+      let messageNlStackTrace = System.String.Join(Environment.NewLine, [ typeName; ": "; message; stackTrace ])
+      use exnLines = new System.IO.StringReader(messageNlStackTrace)
+      let rec go lines =
+        match exnLines.ReadLine() with
+        | null ->
+          List.rev lines // finished reading
+        | line ->
+          if line.StartsWith(stackFrameLinePrefix) then
+            // subtext
+            go ((Environment.NewLine, Text) :: ((line, Subtext) :: lines))
+          else
+            // regular text
+            go ((Environment.NewLine, Text) :: ((line, Text) :: lines))
+      go []
+
+    let literateColorizeExceptions (context : LiterateConsoleConf) message =
+      let exns = Utils.Aether.Lens.getPartialOrElse Message.Lenses.errors_ [] message
+      let getStringFromMapOrFail exnObjMap fieldName =
+        match exnObjMap |> Map.find fieldName with
+        | String m -> m
+        | _ -> failwithf "Couldn't find %s in %A" fieldName exnObjMap
+
+      match exns with
+      | [] -> []
+      | values ->
+        values
+        |> List.choose (function | Object v -> Some v | _ -> None)
+        |> List.map (fun exnObjMap ->
+          let exnTypeName = getStringFromMapOrFail exnObjMap "type"
+          let message = getStringFromMapOrFail exnObjMap "message"
+          let stackTrace = getStringFromMapOrFail exnObjMap "stackTrace"
+          literateExceptionColorizer context exnTypeName message stackTrace
+          @ [ Environment.NewLine, Text ]
+        )
+        |> List.concat
+
+    let rec literateFormatField (options : LiterateConsoleConf)
+                                (prop : Property)
+                                (propValue : Field) =
+      let value, units = match propValue with Field (v, u) -> v, u | _ -> failwith "never happens"
+      match value with
+      | Value.Array items -> [ "todo", OtherSymbol ] // TODO:
+      | Value.String s -> [ s, StringSymbol ]
+      | Value.Bool b -> [ b.ToString(), KeywordSymbol ]
+      | Value.Float f -> [ f.ToString(), NumericSymbol ]
+      | Value.Int64 i -> [ i.ToString(), NumericSymbol ]
+      | Value.BigInt bi -> [ bi.ToString(), NumericSymbol ]
+      | Value.Binary (x, y) -> [ "todo", OtherSymbol ] // TODO:
+      | Value.Fraction (x, y) -> [ "todo", OtherSymbol ] // TODO:
+      | Value.Object o -> [ o.ToString(), OtherSymbol ] // TODO: recurse
```

Comment:
In the `Logary.Formatting` module there's some logic to traverse to form keys like `a.b.c => 3, a.b.d => 55`; the above function would be a nice parameter into that function to avoid having to recurse manually here.